### PR TITLE
Add logging helper functions

### PR DESCRIPTION
**Pull request by rosiemiller2**

### CODE REVIEW

File: mbed_tools_lib/logging.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+"""Helpers for logging errors according to severity of the exception."""
+import logging
+from typing import Any, Optional
+
+
+def log_exception(logger: Optional[Any],
+                  exception: Optional[Exception]) -> None:
```
Comment (madchutney) [inline mbed_tools_lib/logging.py]: Are these really optional?

Comment (rosiemiller2) [reply mbed_tools_lib/logging.py]: Yeah, that doesn't seem to make sense to me, why would you want to log and exception without a logger or an exception?

File: mbed_tools_lib/logging.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    if logger and exception:
+        logger.error(exception)
+        logger.debug(exception, exc_info=True)
+
+
+def set_log_level(verbose_count: int) -> None:
+    """Sets the log level.
+
+    Args:
+        verbose_count: requested log level count
```
Comment (muchzill4) [inline mbed_tools_lib/logging.py]:
```suggestion
        verbose_count: number of `-v` flags used
```

Thoughts?

File: tests/test_logging.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+from mbed_tools_lib.logging import (
+    log_exception,
+    set_log_level,
+    LOGGING_FORMAT,
+)
+
+
+class TestLogException(TestCase):
+    def test_log_exception_error(self):
+        mock_logger = mock.Mock(specsec=logging.Logger)
```
Comment (muchzill4) [inline tests/test_logging.py]: It's `spec_set`.

**Review by muchzill4 — CHANGES_REQUESTED** [review]

File: mbed_tools_lib/logging.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+
+def log_exception(logger: Any, exception: Exception) -> None:
+    """Logs an exception in both normal and verbose forms.
+
+    Args:
+        logger: logger
+        exception: exception to log
+    """
+    if logger and exception:
```
Comment (muchzill4) [inline mbed_tools_lib/logging.py]: If they are no longer optional, do we need this conditional?

File: mbed_tools_lib/logging.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
+"""Helpers for logging errors according to severity of the exception."""
+import logging
+from typing import Any
+
+LOGGING_FORMAT = "%(levelname)s: %(message)s"
+
+
+def log_exception(logger: Any, exception: Exception) -> None:
```
Comment (muchzill4) [inline mbed_tools_lib/logging.py]: Should `logger` be `Any`? 🤔 I'd ask somebody experienced with typing.

Comment (rosiemiller2) [reply mbed_tools_lib/logging.py]: I guess that we need the logger to have the methods `error` and `debug` that expects those args, so it's probably better to restrict to anything that extends `logging.Logger`.

File: tests/test_logging.py (hunk 3 above)
Comment (rosiemiller2) [reply tests/test_logging.py]: 😞

File: mbed_tools_lib/logging.py (hunk 4 above)
Comment (rosiemiller2) [reply mbed_tools_lib/logging.py]: Good point, I don't see why we would.

File: tests/test_logging.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+        log_exception(mock_logger, mock_exception)
+
+        mock_logger.error.assert_called_once_with(mock_exception)
+        mock_logger.debug.assert_called_once_with(mock_exception, exc_info=True)
+
+
+@mock.patch("mbed_tools_lib.logging.logging")
+class TestSetLogLevel(TestCase):
+    def test_debug(self, mocked_logging):
```
Comment (rwalton-arm) [inline tests/test_logging.py:24]: I'm wondering if we should look into using something like [parametrized](https://github.com/wolever/parameterized) to reduce the boilerplate test code. It claims to work with `unittest.TestCase` style tests.

**Review by muchzill4 — APPROVED** [review]
📈

**Review by muchzill4 — APPROVED** [review]
📈

**Comment by muchzill4** [comment]
So good I approved twice.

File: tests/test_logging.py (hunk 6 above)
Comment (rosiemiller2) [reply tests/test_logging.py:24]: I have reduced the boilerplate a little by not having to declare the return value in each test. With that reduction it actually doesn't bother me that much to have a bit of test repetition. I find the test names quick to parse and understand whereas a parametrized test I might need to look over a couple of times to fully grok.

Comment (rosiemiller2) [reply tests/test_logging.py:24]: Good to know that parameterized works with `unittest.TestCase` though.

**Pull request merged**